httpevents/request.go

```diff
@@ -75,5 +75,19 @@ func (r *request) log(logger *events.Logger, depth int) {
 	// transport outside of the httpevents package.
 	l := *logger
 	l.CallDepth += depth + 1
-	l.Log(string(fmt), arg...)
+
+	switch {
+	case is4xx(r.status) || is5xx(r.status):
+		l.Log(string(fmt), arg...)
```

Comment:
Since the methods have the same signature you can achieve this by using a function pointer.
```go
log func(*events.Logger, string, ...interface{})
```
then call it like this:
```go
log(l, "format", args...)
```
It seems similar to having to track a constant to represent whether you want to use `Log` or `Debug`.